Trim blank lines at the end of files on save.

ypid opened the pull request:
Also note that your Vundle installs the latest version of plugins (from GitHub) which might make you vulnerable to an adversary pushing code to one of those repos. I consider my dotfiles a core part of my work environment and try to review it as good as I can. This is the reason why I stopped using Vundle to install plugins and switched to git submodules. Ref: https://github.com/ypid/dotfiles. Maybe you can make the switch before you run `make all` on one of your machines again. I already added the new plugin using git submodules and updated your `Makefile` accordingly. Hope you are alright with that.

Readme: https://github.com/ypid/dotfiles-2/tree/update-2#dotfiles

ypid committed:
Trim blank lines at the end of files on save.

ypid committed:
Updated install documentation and shameless self credits.

ypid committed:
Started installing Vim Plugins via git submodules.

Also note that Vundle installs the latest version AFAIK from GitHub which might make you vulnerable by an adversary pushing code to one of those repos. I consider my dotfiles a core part of my work environment and try to review it as good as I can. This is the reason why I stopped using Vundle to install plugins and switched to git submodules. Ref: https://github.com/ypid/dotfiles. Maybe you can make the switch before you run `make all` on one of your machines again. I already added the new plugin using git submodules and updated your Makefile accordingly. Hope you are alright with that.

drybjed commented:
I've heard that [vim-plug](https://github.com/junegunn/vim-plug) is the hyped vim plugin manager these days. It allows branch/tag cloning, so maybe it's time to switch. My dotfiles are a bit old and I plan to touch them up a bit soon, especially `mutt` configuration, so I'll look into updating my `vim` setup as well. Installing plugins via `vim-plug` with specific tags instead of git submodules should make maintenance easier. Also, thanks for the heads up. :-)

I don't want automatic trimming of every file I open in the editor. For example, [this template](https://github.com/debops/ansible-etc_services/blob/master/templates/etc/services.d/00_ansible.j2) contains empty blank lines at the end to allow some whitespace when it's assembled with the other files into `/etc/services`. With automatic trimming I would probably need to switch to a different editor to make this, and the next time I opened that file in `vim` it would trim it anyway. Awful idea.

ypid committed:
Only remove trailing empty line if there is one empty line at the end.

Addresses valid point by @drybjed.

> I don't want automatic trimming of every file I open in the editor.
> For example, this template contains empty blank lines at the end to
> allow some whitespace when it's assembled with the other files into
> /etc/services. With automatic trimming I would probably need to switch
> to a different editor to make this, and the next time I opened that
> file in vim it would trim it anyway. Awful idea.

ypid commented:
`vim-plug` looks nice. Thanks for the ref. When I need more features for plugin management I know where to look :smile:

> I don't want automatic trimming of every file I open in the editor.

I hope [this](https://github.com/drybjed/dotfiles/pull/2/commits/1624b0358410d7a8167da01e7be601bd0921dea3) addresses your issue.

I just prefer automation, also when that means thinking about a problem a little more in depth than doing repetitive tasks manually.

drybjed commented:
Automation is fine, but to perform it correctly tere needs to be a source of truth, which in our case are various files edited by `vim`. The editor itself shouldn't be a source of truth. In this case your automation changes your source of truth, which might result in unexpected consequences like the one described above. Finding the issue and resolving it if you don't know that your editor messes with your files might take hours.

ypid committed:
Disabled "Trim blank lines at EOF".

Ref: https://github.com/drybjed/dotfiles/pull/2#issuecomment-231705356

ypid commented:
You are right. I disabled this `autocmd`. This issue is too unimportant :wink: Lets concentrate on the important stuff :smile:
